src/storage/LockingResourceStore.ts

```diff
@@ -55,4 +55,24 @@ export class LockingResourceStore implements AtomicResourceStore {
       await lock.release();
     }
   }
+
+  private async lockedRunGet(identifier: ResourceIdentifier, func: () => Promise<Representation>):
```

Comment:
Since this is only used specifically for the GET function, there's no need to pass a function along as parameter. Actually the contents of this function are small enough that I would just put it directly in the `getRepresentation` function.